projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -210,6 +267,645 @@ private static function is_squareish_layout( $attr ) {
+				}
+			}
+
+			$content_parts[] = '</div></div>'; // Close clear:both container and column container
+		}
+
+		$content_parts[] = '</div>'; // Close main flex container
+
+		return self::generate_email_wrapper( implode( '', $content_parts ) );
+	}
+
+	/**
+	 * Build mosaic layout content with flexible row/column structure.
+	 *
+	 * @param array $images Array of image data.
+	 * @param int   $target_width Target width for email.
+	 * @param int   $cell_padding Cell padding.
+	 * @param int   $border_radius Border radius value (0-20).
+	 * @param array $attr Block attributes.
+	 * @return string HTML content.
+	 */
+	private static function build_mosaic_layout_content( $images, $target_width, $cell_padding, $border_radius = 0, $attr = array() ) {
+		$content_parts       = array();
+		$border_radius_style = self::generate_border_radius_style( '', $border_radius );
+
+		// Generate mosaic layout rows
+		$rows = self::generate_mosaic_rows( $images );
+
+		foreach ( $rows as $index => $row ) {
+			// Check if this is the last row
+			$is_last_row   = ( $index === count( $rows ) - 1 );
+			$margin_bottom = $is_last_row ? '0' : $cell_padding;
+
+			$content_parts[] = '<div style="margin-bottom:' . $margin_bottom . 'px;display:flex;width:100%;max-width:100%;box-sizing:border-box">';
+
+			// Calculate width for each image in this row
+			$images_in_row       = count( $row );
+			$total_margin_width  = ( $images_in_row - 1 ) * $cell_padding;
+			$image_width_percent = ( 100 / $images_in_row );
+
+			foreach ( $row as $image_index => $image ) {
+				$margin_right = ( $image_index < count( $row ) - 1 ) ? $cell_padding : '0';
+				$width_style  = 'width:calc(' . $image_width_percent . '% - ' . ( $total_margin_width / $images_in_row ) . 'px);';
+				$image_styles = 'margin:0;width:100%;box-sizing:border-box;';
+
+				$content_parts[] = '<div style="margin:0;margin-right:' . $margin_right . 'px;' . $width_style . 'display:flex;min-width:0;box-sizing:border-box">';
+				$content_parts[] = '<figure style="margin:0;overflow:hidden;padding:0;display:flex;width:100%;box-sizing:border-box">';
+				$content_parts[] = self::generate_image_html( $image, $image_styles, $border_radius_style, $attr );
+				$content_parts[] = '</figure>';
+				$content_parts[] = '</div>';
+			}
+
+			$content_parts[] = '</div>';
+		}
+
+		return self::generate_email_wrapper( implode( '', $content_parts ) );
+	}
+
+	/**
+	 * Generate mosaic layout rows based on image count.
+	 *
+	 * @param array $images Array of image data.
+	 * @return array Array of rows, each containing images.
+	 */
+	private static function generate_mosaic_rows( $images ) {
+		$rows        = array();
+		$image_count = count( $images );
+
+		// More sophisticated mosaic algorithm based on image count
+		if ( $image_count <= 3 ) {
+			// For 3 or fewer images, use simple layout
+			$rows[] = $images;
+		} else {
+			// For more images, create varied row patterns
+			$patterns = array(
```

Comment:
Is it intentional that a pattern for 4 images is not included? It seems like defining such pattern would allow to remove the extra condition in the for-loop `$image_index < $image_count`.